traitlets/traitlets.py

```diff
@@ -489,11 +488,28 @@ def _validate(self, obj, value):
         return value
 
     def _cross_validate(self, obj, value):
-        if hasattr(obj, '_%s_validate' % self.name):
-            cross_validate = getattr(obj, '_%s_validate' % self.name)
-            value = cross_validate(value, self)
+        try:
+            cb = getattr(obj, '_%s_validate' % self.name)
+        except AttributeError:
+            cb = obj._trait_validators.get(self.name,None)
+        else:
+            warn("_[traitname]_validate handlers are deprecated: use register_validator instead",
+                 DeprecationWarning, stacklevel=2)
+            if not callable(cb):
```

Comment:
Typically, in python, rather than checking every error case, we just go ahead and do things, and trust that errors will be raised if there was a problem.  The typical pythonic thing here is to not test to see if cb is callable, but just call the cb function.  If it's not callable, an error will be raised to the user at that point.